src/OutputWriters/netcdf_output_writer.jl

```diff
@@ -13,7 +13,7 @@ using Oceananigans.Diagnostics: WindowedSpatialAverage
 using Oceananigans.Fields: reduced_location, location, FieldSlicer, parent_slice_indices
 
 dictify(outputs) = outputs
-dictify(outputs::NamedTuple) = Dict(string(k) => dictify(v) for (k, v) in zip(keys(outputs), values(outputs)))
+dictify(outputs::NamedTuple) = Dict{Any,Any}(string(k) => dictify(v) for (k, v) in zip(keys(outputs), values(outputs)))
```

Comment:
What issue is this solving?

Reply:
Not sure why the issue pops up since we test for it but when @francispoulin tried to save two function outputs, e.g.

```
(output1 = function1, output2 = function2)
```

`dictify` complained that it couldn't convert `typeof(function2)` to `typeof(function1)` so I suspected that it created a `Dict{String,typeof(function1)}` for some reason.

Explicitly creating a `Dict{Any,Any}` fixes this issue as expected but I don't see why it's necessary...

Reply:
Are we sure there wasn't a bug in the code? If we test it then it should work? Or we can create another test that fails...

Reply:
We looked at it but it looked exactly like what we were doing: https://github.com/CliMA/Oceananigans.jl/blob/ff1131c8ec5f80d563bae59f7967dc0c72a2ebc6/test/test_netcdf_output_writer.jl#L335-L359

Agree that we shouldn't blindly put band-aids in the code. We could keep this PR open until we can put in a proper fix + test.